static/js/user_groups_settings_ui.js

```diff
@@ -95,6 +103,36 @@ function show_right_section() {
     $(".user-groups-header").addClass("slide-left");
 }
 
+export function add_group_to_table(group) {
+    if (is_group_already_present(group)) {
+        // If a group is already listed/added in groups modal,
+        // then we simply return.
+        // This can happen in some corner cases (which might
+        // be backend bugs) where a realm administrator may
+        // get two user_group-add events.
+        return;
```

Comment:
If you can reproduce this, we should debug why the server can generate that -- it's probably the kind of thing that will have other unpleasant side effects :).

Reply:
I don't have a reproducer for this, but I think we should we should do nothing and return in case the group is already present. I had added the comment similar to that present in `stream_settings_ui.js`. https://github.com/zulip/zulip/blob/dbacc00f0f4a4e854d255014b584c58dcdf83d4d/static/js/stream_settings_ui.js#L259-L262

As that appeared like one of the possible cases where we may face that situation, but I have never experienced this while working on it. May be we can update this comment.